docs/streams/stream-basics/producers.md

```diff
@@ -20,7 +24,104 @@ Messages published by producers can be compressed during transportation in order
 
 - [LZ4](https://github.com/lz4/lz4)
 - [ZLIB](https://zlib.net/)
+- [ZSTD](https://facebook.github.io/zstd/)
+- [SNAPPY](https://github.com/google/snappy)
 
 ## Batching
 
-If batching is enabled, then the producer accumulates and sends a batch of messages in a single request. Batching size is defined by the maximum number of messages and maximum publish latency.
\ No newline at end of file
+Macrometa Streams utilize batching to enhance efficiency in message processing. When batching is enabled, the producer aggregates multiple messages and transmits them as a single batched request. The batching size is determined by two factors: the maximum number of messages allowed in a batch and the maximum publish latency.
+
+By default, a batch can accommodate up to 1,000 messages. The default maximum publish latency is set at 10 milliseconds, ensuring a timely transmission of message batches.
+
+## How to create a producer with specific options in SDK
+
+This example shows how to create a producer with specific options in JavaScript SDK.
+When we create a producer, we can specify the following options:
+
+
+|         Option          |                         Description                         | Default |
+| :---------------------: | :---------------------------------------------------------: | :-----: |
+|    sendTimeoutMillis    |                        Send timeout                         | 30 secs |
+|     batchingEnabled     |                 Enable batching of messages                 |  false  |
+|   batchingMaxMessages   |       Maximum number of messages permitted in a batch       |  1000   |
+|   maxPendingMessages    | Set the max size of the internal-queue holding the messages |  1000   |
+| batchingMaxPublishDelay |    Time period within which the messages will be batched    |  10ms   |
+
+<Tabs groupId="modify-single">
+<TabItem value="javascript" label="JavaScript SDK">
+
+
+```js
+const jsc8 = require("jsc8");
+
+const BASE_URL = "https://play.paas.macrometa.io/"
+
+client = new jsc8({
+    url: BASE_URL,
+    apiKey: "xxxxxx",
+    fabricName: "_system",
+});
+
+const streamName = "streamQuickstart";
```

Comment:
Why aren't the constants all together at the beginning? That's the pattern I usually see.